docs/developer/architecture/alerting.rst

```diff
@@ -43,3 +43,295 @@ maintenance operation.
+:doc:`CSC<configurations>` (Cluster and Services Configurations), and use
+the already defined resources for :term:`Alertmanager` and :term:`Prometheus`.
+
+Extra Prometheus Rules
+~~~~~~~~~~~~~~~~~~~~~~
+
+We will use the already existing ``metalk8s-prometheus-config``
+:term:`ConfigMap` to store the :term:`Prometheus` configuration
+customizations.
+
+Adding extra alert and record rules will be done editing this :term:`ConfigMap`
+under the ``spec.extraRules`` key in ``config.yaml`` as follows::
+
+    apiVersion: v1
+    kind: ConfigMap
+    metadata:
+      name: metalk8s-prometheus-config
+      namespace: metalk8s-monitoring
+    data:
+      config.yaml: |-
+        apiVersion: addons.metalk8s.scality.com
+        kind: PrometheusConfig
+        spec:
+          deployment:
+            replicas: 1
+          extraRules:
+            groups:
+            - name: <rulesGroupName>
+              rules:
+              - alert: <AlertName>
+                annotations:
+                  description: description of what this alert is
+                expr: vector(1)
+                for: 10m
+                labels:
+                  severity: critical
+              - alert: <AnotherAlertName>
+                [...]
+              - record: <recordName>
+                [...]
+            - name: <anotherRulesGroupName>
+              [...]
+
+PromQL_ is to be used to define ``expr`` field.
+
+This ``spec.extraRules`` entry will be used to generate through Salt a
+``PrometheusRule`` object named ``metalk8s-prometheus-extra-rules`` in the
+``metalk8s-monitoring`` namespace, which will be automatically consumed by the
+:term:`Prometheus` Operator to generate the new rules.
+
+A CLI and UI tooling will be provided to show and edit this configuration.
+
+.. _PromQL: https://prometheus.io/docs/prometheus/latest/querying/basics/
+
+Edit Existing Prometheus Alert Rules
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To edit existing :term:`Prometheus` rules, we can't only define new
+``PrometheusRules`` resources since :term:`Prometheus` Operator will not
+overwrite those already existing, but will rather append them to the list of
+rules, ending up with 2 rules with the same name but different parameters.
+
+We also can't edit the ``PrometheusRules`` deployed by MetalK8s, otherwise we
+would lose these changes in case of cluster restoration, upgrade or downgrade.
+
+So, in order to allow the user to customize the alert rules, we will pick up
+some of them (the most relevant ones) and expose only few parts of their
+configurations (e.g. threshold) to be customized.
+
+It also makes the customization of these alert rules easier for the user as,
+for example, he will not need to understand PromQL_ to adapt the threshold of
+an alert rule.
+
+Since in :term:`Prometheus` rules, there are duplicated group name + alert rule
+name, we also need to take the severity into account to understand which
+specific alert we're editing.
```

Comment:
I think we are actually taking the `labels` into account since `severity` is a subkey of labels ??

Reply:
No, we specifically take the severity label into account, not the other labels.